libstuff/libstuff.cpp

```diff
@@ -188,6 +188,36 @@ string SStrFromHex(const string& buffer) {
     return retVal;
 }
 
+string SBase32HexStringFromBase32(const string& buffer) {
+    string base32HexString = "0123456789ABCDEFGHIJKLMNOPQRSTUV";
+    string base32String = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567";
+    string newBase32HexString = "";
+
+    for (uint i = 0; i < buffer.length(); i++) {
+        size_t val = base32String.find(buffer[i]);
```

Comment:
This makes a string transliteration run in quadratic time.

What about something like this? This just maps directly from the values in the input to the values we'll have in the output (shifted by 50, since the value for the char `0` is 50, and we don't need anything earlier than than).
```
string SBase32HexStringFromBase32(const string& buffer) {
    static const char map[] = "QRSTUV\0\0\0\0\0\0\0\0\0""0123456789ABCDEFGHIJKLMNOP";
    static const int mapLength = sizeof(map);
    string out = buffer;
    int shiftedIndex;
    for (size_t i = 0; i < out.size(); i++) {
        shiftedIndex = out[i] - 50;
        if (mapLength < shiftedIndex || map[shiftedIndex] == 0) {
            STHROW("Character not found in base32 alphabet.");
        }
        out[i] = map[shiftedIndex];
    }

    return out;
}
```